bitcoin/src/blockdata/transaction.rs

```diff
@@ -1168,6 +1174,29 @@ impl From<&Transaction> for Wtxid {
     fn from(tx: &Transaction) -> Wtxid { tx.wtxid() }
 }
 
+/// Computes the value of an output accounting for the cost of spending it.
+///
+/// The effective value is the value of an output value minus the amount to spend it.  That is, the
+/// effective_value can be calculated as: value - (fee_rate * weight).
+///
+/// Note: the effective value of a [`Transaction`] may increase less than the effective value of
+/// a [`TxOut`] when adding another [`TxOut`] to the transaction.  This happens when the new
+/// [`TxOut`] added causes the output length `VarInt` to increase its encoding length.
+///
+/// # Arguments
+///
+/// * `fee_rate` - `FeeRate` in sat/kwu
```

Comment:
I think the unit is redundant and possibly confusing. The unit is handled by the type and irrelevant here. Rather thand mentioning it and the type (which is seen from signature) I'd prefer to say something like "`fee_rate` - the fee rate of the transaction being created"

Reply:
Alright, changed.  There's a note in the test about the units also, although I think it's helpful.